File: subinterval.cpp
#include <iostream>
#include <cmath>
#include <algorithm>

using namespace std;

int main() {
    // lowest number possible
    int a;
    long double ans = -1.0e100;
    // n
    long int n;
    cin >> n;
    int *array_numbers = new int[n];
    // Getting Numbers
    for (int i = 0; i < n; i++) {
        cin >> array_numbers[i];
    }
    //end of processing input
    int *maximimsum = new int[n];
    maximimsum[0] = array_numbers[0];
    for (int i = 1; i < n; i++)
        maximimsum[i] = fmax(array_numbers[i], array_numbers[i] + maximimsum[i - 1]);
    for (int i = 1; i < n; i++)
        ans = fmax(ans, maximimsum[i]);
    cout << ans << endl;
    return 0;
}

File: python/subinterval.py
n = input()
n = int(n)
ans = -1e100
array_numbers = list(map(int, input().split(' ')))
max_sum = [array_numbers[0]]
for i in range(1, n):
    max_sum.append(max(array_numbers[i], array_numbers[i] + max_sum[i - 1]))
for i in range(n):
    ans = max(ans, max_sum[i])
print(ans)
